frb_example/pure_dart/dart/lib/main.dart

```diff
@@ -183,7 +184,9 @@ void main(List<String> args) async {
     {
       final message = 'Hello there.';
       final ret = await api.handleOptionalStruct(document: message);
-      if (ret == null) fail('handleOptionalStruct returned null for non-null document');
+      if (ret == null) {
+        fail('handleOptionalStruct returned null for non-null document');
+      }
```

Comment:
uber-nit: `expect(ret, isNotNull, reason: ...);` (anyway this is just minor details, no worries)

Reply:
This plays nicer with type promotion so I probably should keep this.

Reply:
Totally agree